### PR TITLE
Recommend DateTimeOffset for databases

### LINKED ISSUE #17909: Do not generally recommend DateTime for Databases
The following is written in the DateTime section

> Retrieve date and time information from sources outside of .NET, such as SQL databases. Typically, these sources store date and time information in a simple format that is compatible with the DateTime structure.

And as a hint:

> When saving or sharing DateTime data, UTC should be used and the DateTime value's Kind property should be set to DateTimeKind.Utc.

The section of DateTimeOffset contains the following:

> These uses for DateTimeOffset values are much more common than those for DateTime values. As a result, DateTimeOffset should be considered the default date and time type for application development.

The problem I keep seeing is that the developers continue to use DateTime and don't care about the timezone.

DateTime has many disadvantages when used in the database, including
- DateTimeOffset is SQL Compliant, DateTime is not
- DateTimeOffset has a date range from 0001 to 9999, DateTime from 1753
- The accuracy of DateTimeOffset is higher

And especially important: DateTimeOffset respects the time zone.

So I think it is appropriate that DateTime is not recommended for direct use to exchange times in APIs, files or databases - but DateTimeOffset.

Since DateTimeOffset is generally the recommended way anyway - it is already mentioned in the documentation - the whole section DateTimeOffset should be moved up.
Otherwise people will only read DateTime and not the very important rest.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7b64d1f3-174c-fffe-d2e6-59a0b62de797
* Version Independent ID: 832e5ebc-6fc0-e596-5827-51034a3f89ab
* Content: [Choosing between DateTime, DateTimeOffset, TimeSpan, and TimeZoneInfo](https://docs.microsoft.com/en-us/dotnet/standard/datetime/choosing-between-datetime#feedback)
* Content Source: [docs/standard/datetime/choosing-between-datetime.md](https://github.com/dotnet/docs/blob/master/docs/standard/datetime/choosing-between-datetime.md)
* Product: **dotnet**
* Technology: **dotnet-standard**
* GitHub Login: @Thraka
* Microsoft Alias: **adegeo**

### PR DESCRIPTION
**Pull request by tonytins**
## Summary

- Moved DateTimeOffset above DateTime
- ~~Added compliance with ISO 8601 standard to DateTimeOffset list~~

Fixes #17909

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=19627) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=19627) before we can accept your contribution.<br/><br/>:x: tonytins [sign now](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=19627)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/docs?pullRequest=19627) it.</sub>

**Comment by IEvangelist** [comment]
Hi @tonytins any updates on this draft, intentions of promoting it to an actual PR?

**Comment by tonytins** [comment]
Yes. Sorry. I didn't mean to put this in the draft for so long.

**Comment by adegeo** [comment]
@tarekgh i believe this all good, can you just double check? thanks!

File: docs/standard/datetime/choosing-between-datetime.md [hunk 1]
```diff
@@ -38,6 +38,30 @@ ms.assetid: 07f17aad-3571-4014-9ef3-b695a86f3800
...
+
+The <xref:System.DateTimeOffset> structure represents a date and time value, together with an offset that indicates how much that value differs from UTC. Thus, the value always unambiguously identifies a single point in time.
+
+The <xref:System.DateTimeOffset> type includes all of the functionality of the <xref:System.DateTime> type along with time zone awareness. This makes it suitable for applications that:
+
+- Uniquely and unambiguously identify a single point in time. The <xref:System.DateTimeOffset> type can be used to unambiguously define the meaning of "now", to log transaction times, to log the times of system or application events, and to record file creation and modification times.
+
+- Perform general date and time arithmetic.
+
+- Compliant with ANSI and ISO 8601 standards.
```
Comment (tarekgh) [inline docs/standard/datetime/choosing-between-datetime.md:51]: what exactly you mean by that?

Comment (tarekgh) [reply docs/standard/datetime/choosing-between-datetime.md:51]: I think this will be confusing statement. DateTimeOffset support formatting with 'O' and 'r' for round tripping and RFC1123 which is close to ISO 8601 but not exact. Also, DateTimeFormat support many other format than ISO 8601. https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-date-and-time-format-strings.

Comment (tonytins) [reply docs/standard/datetime/choosing-between-datetime.md:51]: This is why it was still draft and unfortunately fell to the way side. I had hard time with this section.

Comment (tarekgh) [reply docs/standard/datetime/choosing-between-datetime.md:51]: ok, please remove this statement then.

Comment (tonytins) [reply docs/standard/datetime/choosing-between-datetime.md]: Done.

File: docs/standard/datetime/choosing-between-datetime.md [hunk 2]
```diff
@@ -63,28 +83,6 @@ Unless a particular <xref:System.DateTime> value represents UTC, that date and t
...
 
-## The DateTimeOffset structure
-
-The <xref:System.DateTimeOffset> structure represents a date and time value, together with an offset that indicates how much that value differs from UTC. Thus, the value always unambiguously identifies a single point in time.
-
-The <xref:System.DateTimeOffset> type includes all of the functionality of the <xref:System.DateTime> type along with time zone awareness. This makes it suitable for applications that:
-
-- Uniquely and unambiguously identify a single point in time. The <xref:System.DateTimeOffset> type can be used to unambiguously define the meaning of "now", to log transaction times, to log the times of system or application events, and to record file creation and modification times.
-
-- Perform general date and time arithmetic.
```
Comment (tarekgh) [inline docs/standard/datetime/choosing-between-datetime.md:74]: why did you remove this line?

Comment (tonytins) [reply docs/standard/datetime/choosing-between-datetime.md:74]: That was unintended.

Comment (tarekgh) [reply docs/standard/datetime/choosing-between-datetime.md:74]: thanks @tonytins

**Review by tarekgh — APPROVED** [review]

**Comment by adegeo** [comment]
Thanks everyone!
